src/Roslyn.Suppressions/Roslyn.Suppressions/SonarDiagnosticSuppressor.cs

```diff
@@ -0,0 +1,40 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace SonarLint.VisualStudio.Roslyn.Suppressions
+{
+    /// <summary>
+    /// Diagnostic suppressor that can suppress all Sonar C# and VB.NET diagnostics
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    internal class SonarDiagnosticSuppressor : DiagnosticSuppressor
+    {
+        public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions => SupportedSuppressionsBuilder.Instance.Descriptors;
+
+        public override void ReportSuppressions(SuppressionAnalysisContext context)
+        {
+            // TODO - implement
```

Comment:
The SonarCloud code coverage check is failing because this method isn't covered.

The generated code isn't included in the coverage numbers, so as far as SonarCloud is concerned the only new lines of code are the ones in this file.
There are no tests for this empty method -> 50% coverage.

I'm didn't add a dummy test just to shut SonarCloud up; we'll add real tests soon enough.